### DISCUSSION TITLE
OPENFAST for marine turbine environment

### DISCUSSION THREAD

**Discussion by sdragonslayer** [OpenFAST/openfast#2331] [Q&A] [answered]
I was recently reading papers by NREL , specifically this one (https://www.nrel.gov/docs/fy24osti/87249.pdf). I then looked at their code (https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/MHK_RM1_Floating), but realized that their simulation for ocean currents just used the inflow wind module for the underwater currents that turned the turbine. 

Is that sufficient to model underwater conditions, and is there a better alternative to reproduce some of their research?

**Comment by willwiley** [comment, marked as answer]
Hello,

Inflow wind does work for marine currents. The standard options for shear profiles or turbulence spectra (if you want to do this with TurbSim) might not be a good fit. You have the option to input user defined profiles and spectra if you have that information available. 

I believe the response plot shown in the paper you mentioned was for a load case with a steady uniform inflow current with no shear. The inflow wind input file in r-test does have shear defined. At this point, the current defined in HydroDyn should match the current defined in inflow wind, but on r-test the HydroDyn current has no shear. This will be corrected to match. 

Another difference is when waves are present, the wave particle velocities can also influence the rotor inflow. Superposition of the wave and inflow current velocities are coming soon, but are not currently available. 

Thanks

**Comment by sdragonslayer** [comment]
Thank you !
